api/dps/server.go

```diff
@@ -31,20 +32,21 @@ import (
 // This is generally an on-disk interface, but could be a GRPC-based index as
 // well, in which case there is a double redirection.
 type Server struct {
-	index dps.Reader
-	codec dps.Codec
-
+	index    dps.Reader
+	codec    dps.Codec
+	tracer   *trace.Tracer
 	validate *validator.Validate
 }
 
 // NewServer creates a new server, using the provided index reader as a backend
 // for data retrieval.
-func NewServer(index dps.Reader, codec dps.Codec) *Server {
+func NewServer(index dps.Reader, codec dps.Codec, tracer *trace.Tracer) *Server {
 
 	s := Server{
 		index:    index,
 		codec:    codec,
 		validate: validator.New(),
+		tracer:   tracer,
```

Comment:
doesn't look like `tracer` is used. is there something that needs to happen in the server to inject it?

Reply:
yeah per @SaveTheRbtz I have to roll my own, the `flow-go` tracer functions aren't going to cut it for our use-case